collector/cpu_linux.go

```diff
@@ -424,4 +426,10 @@ func (c *cpuCollector) updateCPUStats(newStats map[int64]procfs.CPUStat) {
 
 		c.cpuStats[i] = cpuStats
 	}
+
+	// Remove offline CPUs.
+	onlineCPUIds := maps.Keys(newStats)
```

Comment:
Would it make sense to add a pre-check for this to avoid expensive maps/slices operations?

```
if len(newStats) != len(c.cpuStats) {
...
}
```

Reply:
absolutely, the precheck has been added in new commit.